client/src/components/InterviewCard/InterviewCard.jsx

```diff
@@ -3,54 +3,61 @@ import InterviewCardCSS from "./InterviewCard.module.css";
 import QuestionBlock from "./QuestionBlock";
 import baseurl from "../../url.js";
 import { toast } from "react-toastify";
+import useRatingStore from "../../store/ratingStore";
+import useUserStore from "../../store/userStore";
+import { Navigate, useNavigate } from "react-router-dom";
 
 const InterviewCard = () => {
   const routeHTTP = `${baseurl}`;
 
-  let interviewData = {
-    interviewee: "Jon Snow",
-    attempt: 1,
-    currentScore: 0,
-    questions: [
-      {
-        questNum: 1,
-        title: "Sort Array",
-        content:
-          "Given an array of numbers, write a function that sorts the number from smallest to largest.",
-        note: "",
-      },
-      {
-        questNum: 2,
-        title: "Tree in the woods",
-        content:
-          "If a tree falls in the woods and nobody is around to hear it, what is the circumference of the sun?",
-        note: "",
-      },
-      {
-        questNum: 3,
-        title: "Apples and oranges",
-        content:
-          "Jimmy has one apple and Susie has two apples, how many oranges do they have together?",
-        note: "",
-      },
-    ],
+  //importing state from ratingStore(zustand)
+  const [note, setNote] = useState("");
+  const [step, setStep] = useState(1);
+  const rating1 = useRatingStore((state) => state.rating1);
+  const rating2 = useRatingStore((state) => state.rating2);
+  const rating3 = useRatingStore((state) => state.rating3);
+  const setRating1 = useRatingStore((state) => state.setRating1);
+  const setRating2 = useRatingStore((state) => state.setRating2);
+  const setRating3 = useRatingStore((state) => state.setRating3);
+  const studentId = useUserStore((state) => state.studentId);
+  const setAttempts = useUserStore((state) => state.setAttempts);
+  const attempts = useUserStore((state) => state.attempts);
+  const studentName = useUserStore((state) => state.studentName);
+  const userId = parseInt(localStorage.getItem("userid"));
+  const navigate = useNavigate();
+
+  const setAttemptNumber = () => {
+    if (!studentId) {
+      console.error("StudentId not generated yet");
+      return;
+    }
```

Comment:
navigate back to dashboard instead here if no student is selected. remove interview link from sidebar nav.